implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -52,10 +52,54 @@ async fn run_impl(
     ctx: Context,
     (opts, cmd): (CommandGlobalOpts, ShowCommand),
 ) -> miette::Result<()> {
-    let node_name = get_node_name(&opts.state, &cmd.node_name);
+    if !opts.terminal.can_ask_for_user_input() {
+        show_node(&opts, &ctx, &get_node_name(&opts.state, &cmd.node_name)).await?;
+        return Ok(());
+    }
+
```

Comment:
We forgot about one case here, after this `if`:

```rust
if let Some(node_name) = &cmd.node_name {
    show_node(&opts, &ctx, node_name).await?;
    return Ok(());
}
```

Perhaps we could even merge both, since, if the user passes the `node_name` arg, then we want to show it without any interactivity:

```rust
if cmd.node_name.is_some() || !opts.terminal.can_ask_for_user_input() {
    show_node(&opts, &ctx, &get_node_name(&opts.state, &cmd.node_name)).await?;
    return Ok(());
}
```

Reply:
fixed in [aa2ff2f](https://github.com/build-trust/ockam/pull/6665/commits/aa2ff2f392027e321082531c603b5644eb5372bf)